### PR TITLE
[FEAT] - SplitAbstain vote & Remove Vote

### PR DESCRIPTION
**Pull request by UrbanWill**
# Description

## ⛑️ **What was done:**

- Added SplitAbstain vote functionality
- Display pending votes that are not yet concluded
- Change and Remove votes of ongoing referendums

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Ready to merge

### 🧪 **How to test:**
- SplitAbstain Vote functionality:

1. Visit `/voting`
2. Click on "Abstain"
3. Select a proposal
4. Fill in amounts to Abstain
5. Click review
6. Sign transaction with all required signers

Expected behavior: 
- Transaction should go through, transaction details should be displayed correctly on side sheet on every transaction status
- Transaction should be displayed on Pending Votes list on `/votes`
- Only the latest vote should be displayed on Pending votes list

- Remove Vote functionality:
1. Visit `/voting`
2. If the multisig doesn't have a Aye, Nay or Abstain transaction, follow the SplitAbstain Vote functionality steps above
3. Click "Remove" on Pending votes list
4. Sign transaction with required signers

Expected behaviour:
- Transaction should go through and displayed on `/overview`
- Vote should no longer be displayed on Pending votes list

### 🗒️ **Notes:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### 🖼️ **Screenshots:**

https://github.com/TalismanSociety/signet-web/assets/47801291/7e4cce0b-5fe8-4d21-af35-040c1318fdbb

### CODE REVIEW

**Review by chrisling-dev — APPROVED** [review]
LGTM after comments

File: apps/multisig/src/domains/referenda/index.ts [hunk 1]
```diff
@@ -69,12 +81,14 @@ export const useReferenda = (chain: Chain) => {
...
 
   const getReferendums = useCallback(async () => {
+    setIsLoading(true)
     if (apiLoadable.state !== 'hasValue' || isPalletSupported === undefined) return
 
     if (!isPalletSupported) {
       console.error(`referenda or conviction_voting pallets not supported on this chain ${chain.chainName}`)
       // treat it as 0 referendum created if required pallets are not supported
       setReferendums([])
+      setIsLoading(false)
```
[the remark this answers is not in the document]
Comment (UrbanWill) [reply apps/multisig/src/domains/referenda/index.ts:91]: You are absolutely right! `useReferenda` is a great candidate for a react-query refactor for multiple reasons. IMO specially for what you just mentioned and caching. I didn't want to scope creep...

Here is a blogpost highlighting some of the problems it solves, number 2 is handling loading state: https://tkdodo.eu/blog/why-you-want-react-query

File: apps/multisig/src/domains/referenda/index.ts [hunk 2]
```diff
@@ -98,17 +113,20 @@ export const useReferenda = (chain: Chain) => {
...
+  return { referendums, isPalletSupported, isLoading }
 }
 
 export const isVoteDetailsComplete = (voteDetails: VoteDetails) => {
   if (voteDetails.referendumId === undefined) return false
 
-  if (voteDetails.details.Standard) {
-    const { balance } = voteDetails.details.Standard
+  if (voteDetails.convictionVote === 'Standard') {
+    const { balance } = voteDetails.details.Standard!
```
[the remark this answers is not in the document]
Comment (UrbanWill) [reply apps/multisig/src/domains/referenda/index.ts]: A note here is that now we have two vote types coexisting, it is required to check `convictionVote` as one form may be filled but not the other.

The issue was that we were using `VoteDetails` type for both the voting form, where we need all voting types at all times, and extrinsicToDecoded function, where it *really* is optional. 

The solution is to create a `VoteDetailsForm` where Standard,  SplitAbstain and Split are non-optional and remove non-null asserts and optional chaining :)

File: apps/multisig/src/layouts/NewTransaction/Vote/ConvictionsDropdown.tsx [hunk 3]
```diff
@@ -20,16 +21,25 @@ export function createConvictionsOpts(): { headlineText: string; value: number }
...
 
 // ref: https://github.com/polkadot-js/apps/blob/master/packages/react-components/src/ConvictionDropdown.tsx
-const ConvictionsDropdown: React.FC<Props> = ({ conviction, onChange }) => {
+const ConvictionsDropdown: React.FC<Props> = ({ conviction, setVoteDetails }) => {
   const options = createConvictionsOpts()
+
+  const handleChange = (value: number) => {
+    setVoteDetails(prev => {
+      const updatedVal = { ...prev }
+      updatedVal.details.Standard!.vote.conviction = value
```
[the remark this answers is not in the document]
Comment (UrbanWill) [reply apps/multisig/src/layouts/NewTransaction/Vote/ConvictionsDropdown.tsx]: Updated types as mentioned above and removed non-null assertions, we safe!

**Pull request merged**